frontend/app/routes/_app+/letters+/_index.tsx

```diff
@@ -40,7 +40,7 @@ export async function loader({ request }: LoaderFunctionArgs) {
   const interopService = getInteropService();
   const userId = await userService.getUserId();
   const letters = await interopService.getLetterInfoByClientId(userId, 'clientId', sortOrder); // TODO where and what is clientId?
-  const letterTypes = (await interopService.getAllLetterTypes()).filter(({ code }) => letters.some(({ name }) => name === code));
+  const letterTypes = (await interopService.getAllLetterTypes()).filter((code) => letters.map((letter) => letter.name).includes(code.code));
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

```suggestion
  const letterTypes = (await interopService.getAllLetterTypes()).filter(({ code }) => letters.some((letter) => letter.name === code));
```